### PR TITLE
Add work command

### PR DESCRIPTION
**Pull request by ToothyDev**
Includes several jobs with an upper and lower pay bracket, failure chances and custom failure messages for each job. The option inside discord gets autocompleted so all jobs are visible to the user.
I could add picking a random job in case the user doesn't specify a job to use, please let me know

### CODE REVIEW

File: cogs/currency.py [hunk 1]
```diff
@@ -68,6 +70,23 @@ async def withdraw(self, ctx, amount):
...
+        if job.title() not in jobs.Job.joblist:
+            return await ctx.respond("Please select a valid job!", ephemeral=True)
+        work = jobs.Job(job)
+        if random.randint(0, 100) > work.success_chance:  # If job FAILS
+            return await ctx.respond(f"{utilities.data.Emotes.crossmark} {work.fail_message}")
+        else:
+            pay = random.randint(work.min_pay, work.max_pay)
+            if not await selector('SELECT * FROM economy WHERE UID = %s', [ctx.author.id]):
+                await modifyData("INSERT INTO economy (UID,CASH, BANK) VALUES(%s, 0, 0)", [ctx.author.id])
+            await modifyData('UPDATE economy SET CASH = CASH + %s WHERE UID = %s', [pay, ctx.author.id])
```
Comment (JoshuaSlui) [inline cogs/currency.py]: Please use `ON DUPLICATE KEY` instead. That would spare you two unnecessary database calls.

File: cogs/currency.py [hunk 2]
```diff
@@ -68,6 +70,23 @@ async def withdraw(self, ctx, amount):
...
+    @slash_command()
+    @option("job", str, description="The job you want to work", required=True, autocomplete=jobs.Job.autocomplete)
+    @commands.cooldown(1, 60, commands.BucketType.user)
+    async def work(self, ctx, job):
+        """ Work a job """
+        if job.title() not in jobs.Job.joblist:
+            return await ctx.respond("Please select a valid job!", ephemeral=True)
+        work = jobs.Job(job)
+        if random.randint(0, 100) > work.success_chance:  # If job FAILS
+            return await ctx.respond(f"{utilities.data.Emotes.crossmark} {work.fail_message}")
```
Comment (JoshuaSlui) [inline cogs/currency.py]: Please use `Emotes.crossmark`.

File: cogs/currency.py [hunk 3]
```diff
@@ -4,6 +4,8 @@
 from discord.ext import commands
 from utilities.database import selector, createCooldown, checkCooldown, modifyData
 from utilities.data import Colors, Emotes
+import jobs
+import utilities.data
```
Comment (JoshuaSlui) [inline cogs/currency.py]: This was already imported in a more efficient manner on line 6.

File: cogs/currency.py [hunk 4]
```diff
@@ -68,6 +70,23 @@ async def withdraw(self, ctx, amount):
...
+            return await ctx.respond("Please select a valid job!", ephemeral=True)
+        work = jobs.Job(job)
+        if random.randint(0, 100) > work.success_chance:  # If job FAILS
+            return await ctx.respond(f"{utilities.data.Emotes.crossmark} {work.fail_message}")
+        else:
+            pay = random.randint(work.min_pay, work.max_pay)
+            if not await selector('SELECT * FROM economy WHERE UID = %s', [ctx.author.id]):
+                await modifyData("INSERT INTO economy (UID,CASH, BANK) VALUES(%s, 0, 0)", [ctx.author.id])
+            await modifyData('UPDATE economy SET CASH = CASH + %s WHERE UID = %s', [pay, ctx.author.id])
+            return await ctx.respond(f"{utilities.data.Emotes.checkmark} You did your job well!\nPay: {pay} {utilities.data.Emotes.cash}")
```
Comment (JoshuaSlui) [inline cogs/currency.py]: Please use `Emotes.checkmark` and `Emotes.cash`.

**Review by JoshuaSlui — CHANGES_REQUESTED** [review]

File: cogs/currency.py (hunk 1 above)
Comment (ToothyDev) [reply cogs/currency.py]: Not quite sure how to use that in this scenario. Would I need to always run the second line, and then add ON DUPLICATE KEY to actually modify the cash if the user already has an entry?

Comment (JoshuaSlui) [reply cogs/currency.py]: You can see usage of it in the `daily` command.

**Comment by JoshuaSlui** [comment]
Please mind the merge conflict too^

**Review by JoshuaSlui — CHANGES_REQUESTED** [review]
Move jobs.py to the utilities folder and we're good.

**Review by JoshuaSlui — CHANGES_REQUESTED** [review]
Move jobs.py to the utilities folder and we're good.

**Review by JoshuaSlui — APPROVED** [review]

**Pull request merged**